readiness/controller/pkg/conditions/resourceexistence_test.go

```diff
@@ -164,9 +185,71 @@ var _ = Describe("Readiness controller", func() {
 	})
 
 	It("should fail when resourceExistenceCondition is undefined", func() {
-		state, msg := NewResourceExistenceConditionFunc(dynamicClient, discoveryClient)(context.TODO(), nil, "undefinedCondition")
+		state, msg := NewResourceExistenceConditionFunc()(context.TODO(), queryClient, nil, "undefinedCondition")
 
 		Expect(state).To(Equal(corev1alpha2.ConditionFailureState))
 		Expect(msg).To(Equal("resourceExistenceCondition is not defined"))
 	})
+
+	It("should succeed when query client has required permissions", func() {
+		customQueryClient, err := getCustomQueryClient()
+		Expect(err).To(BeNil())
+		Expect(customQueryClient).ToNot(BeNil())
+
+		newPod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-pod",
+				Namespace: defaultNamespace,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "test-container",
+						Image: "test:tag",
+					},
+				},
+			},
+		}
+
+		err = k8sClient.Create(context.TODO(), &newPod)
+		Expect(err).To(BeNil())
+
+		state, msg := NewResourceExistenceConditionFunc()(context.TODO(), customQueryClient, &corev1alpha2.ResourceExistenceCondition{
+			APIVersion: "v1",
+			Kind:       "Pod",
+			Namespace:  &newPod.Namespace,
+			Name:       newPod.Name,
+		},
+			"podCondition")
+
+		Expect(state).To(Equal(corev1alpha2.ConditionSuccessState))
+		Expect(msg).To(Equal("resource found"))
+	})
+
+	It("should fail when query client is missing required permissions", func() {
+		customQueryClient, err := getCustomQueryClient()
+		Expect(err).To(BeNil())
+		Expect(customQueryClient).ToNot(BeNil())
+
+		state, _ := NewResourceExistenceConditionFunc()(context.TODO(), customQueryClient, &corev1alpha2.ResourceExistenceCondition{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+			Name:       "test-deploy",
+			Namespace: func() *string {
+				n := defaultNamespace
+				return &n
+			}(),
+		},
+			"deploymentCondition")
+
+		Expect(state).To(Equal(corev1alpha2.ConditionFailureState))
```

Comment:
nit: can you also assert the message here?

Reply:
Counterpoint: I would skip asserting string messages unless absolutely necessary. It's a bad idea since a message change will break tests. Generally, consumers don't care about the exact string as long as state is being conveyed in a separate field (like success/failure).

So I suggest removing all string asserts.

Reply:
Removed assertions for `message` field